providers/base/bin/reboot_check_test.py

```diff
@@ -0,0 +1,486 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import subprocess
+import re
+import shutil
+import filecmp
+import sys
+import enum
+import typing as T
+from checkbox_support.scripts.image_checker import has_desktop_environment
+
+
+# Checkbox could run in a snap container, so we need to prepend this root path
+RUNTIME_ROOT = os.getenv("CHECKBOX_RUNTIME", default="")
+# Snap mount point, see
+# https://snapcraft.io/docs/environment-variables#heading--snap
+SNAP = os.getenv("SNAP", default="")
+
+
+class ShellResult:
+    """
+    Wrapper class around the return value of run_command, guarantees non-null
+    """
+
+    def __init__(self, return_code: int, stdout: str, stderr: str):
+        self.return_code = return_code
+        self.stdout = stdout
+        self.stderr = stderr
+
+
+def run_command(args: T.List[str]) -> ShellResult:
+    """Wrapper around subprocess.run
+
+    :param args: same args that goes to subprocess.run
+    :type args: T.List[str]
+    :return: return code, stdout and stderr, all non-null
+    :rtype: ShellResult
+    """
+    # PIPE is needed for subprocess.run to capture stdout and stderr
+    # <=3.7 behavior
+    try:
+        out = subprocess.run(
+            args, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+    except FileNotFoundError as e:
+        return ShellResult(1, "", "Command {} not found {}".format(args[0], e))
+
+    return ShellResult(
+        return_code=out.returncode,
+        # if there's nothing on stdout, .stdout is None (<=3.7 behavior)
+        # so we need a default value
+        stdout=(out.stdout or b"").decode(),
+        stderr=(out.stderr or b"").decode(),
+    )
+    # This could throw on non-UTF8 decodable byte strings
+    # but that should be rare since utf-8 is backwards compatible with ascii
```

Comment:
Thanks for reviewing! I had that wrapper in there initially because of the comment on line 52, which turned out to be false since there's always a `b""` in stdout/stderr if we specify `stdout=PIPE` and `stderr=PIPE`. 

But regarding the return_code issue I don't think we really want to end the script if we see a non-zero return code. The original shell script tries to run all the tests even if one of them fails to at least get a result. Do you want to preserve this behavior?